stixcore/processing/stix-pipline

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import shutil
+import logging
+import argparse
+from enum import Enum
+from pathlib import Path
+from collections import defaultdict
+
+from stixcore.config.config import CONFIG
+from stixcore.ephemeris.manager import Spice, SpiceKernelManager
+from stixcore.io.soc.manager import SOCManager
+from stixcore.processing.L0toL1 import Level1
+from stixcore.processing.L1toL2 import Level2
+from stixcore.processing.LBtoL0 import Level0
+from stixcore.processing.TMTCtoLB import process_tmtc_to_levelbinary
+from stixcore.soop.manager import SOOPManager
+from stixcore.tmtc.packets import TMTC
+
+
+def clear_dir(dir):
+    for files in os.listdir(dir):
+        path = os.path.join(dir, files)
+        try:
+            shutil.rmtree(path)
+        except OSError:
+            os.remove(path)
+
+
+class ProductLevel(Enum):
+    TM = -2
+    LB = -1
+    L0 = 0
+    L1 = 1
+    L2 = 2
+    ALL = 100
+
+    def __str__(self):
+        return self.name
+
+    @staticmethod
+    def from_str(label):
+        label = label.upper()
+        for e in ProductLevel:
+            if e.name == label:
+                return e
+        return ProductLevel.TM
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='stix pipeline processing')
```

Comment:
I guess this is batch processing rather than online?

Reply:
currently this is in batch mode
would you prefer a file by file option as well?